www/js/diary/list.js

```diff
@@ -622,43 +636,63 @@ angular.module('emission.main.diary.list', ['ui-leaflet',
     };
 
     var closePurposePopover = function ($event, isOther) {
-      $scope.chosen.purpose = '';
+      $scope.selected.purpose = {
+        label: ''
+      };
       if (isOther == false)
         $scope.draftPurpose = angular.undefined;
       Logger.log("in closePurposePopover, setting draftPurpose = " + JSON.stringify($scope.draftPurpose));
       $scope.purposePopover.hide($event);
     };
 
+    /**
+     * Store selected value for options
+     */
+    $scope.selected = {
+      mode: {
+        label: ''
+      },
+      purpose: {
+        label: ''
+      }
+    };
+    /**
+     * Store currently chosen value
```

Comment:
Can you add some more documentation on `chosen` versus `selected`? Is the thought that the user first chooses the mode and then does something else to `select` it? Or vice versa?

Reply:
`selected` is for display purpose only, this is to display the currently selected option on the list.

`chosen` is for storing the actual value.

The previous code was like
```
$scope.mode = '';
$scope.purpose = '';
$scope.chosen = {
  mode: '',
  purpose: '',
}
```

Now I am refactor it to be more appropriate by moving `$scope.mode` and `$scope.purpose` to `$scope.chosen`. And `$scope.chosen` to `$scope.selected` for readability and structure.

Reply:
ok, but can you add this as a comment to the code? Not everybody is going to go and find the pull request related to a change.

Reply:
concretely, this line

> selected is for display purpose only, this is to display the currently selected option on the list.

Reply:
Sure, done.